agent/exploits/cve_2024_31461.py

```diff
@@ -0,0 +1,95 @@
+"""Agent Asteroid implementation for CVE-2024-31461"""
+
+import requests
+from requests import exceptions as requests_exceptions
+from ostorlab.agent.mixins import agent_report_vulnerability_mixin
+from ostorlab.agent.kb import kb
+
+
+from agent import definitions
+from agent import exploits_registry
+
+VULNERABILITY_TITLE = "SSRF flaw in Plane Project Management Software"
+VULNERABILITY_REFERENCE = "CVE-2024-31461"
+VULNERABILITY_DESCRIPTION = (
+    "Plane, an open-source project management tool, has a Server-Side Request Forgery (SSRF) vulnerability in versions prior to 0.17-dev. This "
+    "issue may allow an attacker to send arbitrary requests from the server hosting the application, potentially leading to unauthorized access to internal systems. "
+)
+
+DEFAULT_TIMEOUT = 30
+MAX_REDIRECTS = 2
+
+
+@exploits_registry.register
+class CVE202431461Exploit(definitions.Exploit):
+    """
+    CVE-2024_31461: SSRF flaw in Plane Project Management Software.
+    """
+
+    def accept(self, target: definitions.Target) -> bool:
+        session = requests.Session()
+        session.max_redirects = MAX_REDIRECTS
+        session.verify = False
+
+        try:
+            resp = session.get(target.url, timeout=DEFAULT_TIMEOUT)
+            if resp.status_code == 200 and "Plane" in resp.text:
+                return True
+
+        except requests_exceptions.RequestException:
+            return False
+
+        return False
+
+    def check(self, target: definitions.Target) -> list[definitions.Vulnerability]:
+        session = requests.Session()
+        session.max_redirects = MAX_REDIRECTS
+        session.verify = False
+
+        vulnerabilities: list[definitions.Vulnerability] = []
+
+        try:
+            resp = session.get(target.url, timeout=DEFAULT_TIMEOUT)
+            if resp.status_code == 200 and "EmojiPickerReact" not in resp.text:
```

Comment:
Can you please add a comment justifying why are you looking for this string.

Reply:
done